lib/logstash/outputs/syslog.rb

```diff
@@ -165,8 +165,20 @@ def publish(event, payload)
     end
 
     if @is_rfc3164
-      timestamp = event.sprintf("%{+MMM dd HH:mm:ss}")
-      syslog_msg = "<#{priority.to_s}>#{timestamp} #{sourcehost} #{appname}[#{procid}]: #{message}"
+      timestamp_tmp = event.sprintf("%{+MMM dd HH:mm:ss}")
+      
+      if timestamp_tmp[4].to_i == 0
+        timestamp = event.sprintf("%{+MMM  d HH:mm:ss}") 
+      else
+        timestamp = event.sprintf("%{+MMM d HH:mm:ss}") 
+      end
+      
+      if @procid
```

Comment:
I suggest to change this in a way, where the `syslog_msg` line is not (nearly) completely duplicated.